### PR TITLE
 Ent - IngestPkgEquals with tests

### PR DESCRIPTION
**Pull request by mrizzi**
# Description of the PR

Follow up on https://github.com/guacsec/guac/pull/1239

The only difference is in the `Query on pkg algo and pkg` where I added `Subpath: ptrfrom.String("")` to the specs otherwise the query would have returned both the ingested `pkgEqual`, basically having Ent backend to consider the lack of a field in the query's spec meaning "don't filter by this field"

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/backends/ent/backend/pkgequal_test.go [hunk 1]
```diff
@@ -544,6 +544,209 @@ func (s *Suite) TestPkgEqual() {
...
+						{
+							Justification: "test justification",
+						},
+						{
+							Justification: "test justification",
+						},
+					},
+				},
+			},
+			Query: &model.PkgEqualSpec{
```
Comment (pxp928) [inline pkg/assembler/backends/ent/backend/pkgequal_test.go:663]: hmm is this a bug on the `inmem` backend that it does not return all if the `subpath` is not set?

Comment (mrizzi) [reply pkg/assembler/backends/ent/backend/pkgequal_test.go:663]: I think so based on consideration that lacking a field in the query's spec should mean "don't filter by this field"

Comment (mrizzi) [reply pkg/assembler/backends/ent/backend/pkgequal_test.go:663]: Draft PR to reproduce the issue https://github.com/guacsec/guac/pull/1249

Comment (pxp928) [reply pkg/assembler/backends/ent/backend/pkgequal_test.go:663]: yeah let's discuss this there. No need to block this PR.

**Review by jeffmendoza — APPROVED** [review]

**Pull request merged**